src/Fixer/PhpUnit/PhpUnitNamespacedFixer.php

```diff
@@ -81,10 +86,41 @@ public function configure(array $configuration = null)
 
         if (PhpUnitTargetVersion::fulfills($this->configuration['target'], PhpUnitTargetVersion::VERSION_6_0)) {
             $this->originalClassRegEx = '/^PHPUnit_\w+$/i';
+            $this->originalClassTranslations = [
+                'PHPUnit_Extensions_PhptTestCase' => 'PHPUnit_Runner_PhptTestCase',
+                'PHPUnit_Framework_Constraint' => 'PHPUnit_Framework_Constraint_Constraint',
+                'PHPUnit_Framework_Constraint_StringMatches' => 'PHPUnit_Framework_Constraint_StringMatchesFormatDescription',
+                'PHPUnit_Framework_Constraint_JsonMatches_ErrorMessageProvider' => 'PHPUnit_Framework_Constraint_JsonMatchesErrorMessageProvider',
+                'PHPUnit_Framework_Constraint_PCREMatch' => 'PHPUnit_Framework_Constraint_RegularExpression',
+                'PHPUnit_Framework_Constraint_ExceptionMessageRegExp' => 'PHPUnit_Framework_Constraint_ExceptionMessageRegularExpression',
+                'PHPUnit_Framework_Constraint_And' => 'PHPUnit_Framework_Constraint_LogicalAnd',
```

Comment:
if I get this right, `PHPUnit_Framework_Constraint_LogicalAnd`  was never existing. It was introduced alread with `\` (`PHPUnit\Framework\Constraint\LogicalAnd`).

for that, currently, your proposal:
- takes original name
- translates it to fake name that would follow transition pattern
- applies transition pattern

we can simplify it that this table can contain already ready to use name with `\`, and then:
- if original name exists in dictionary, use name from dictionary
- if not, apply transition pattern

Reply:
This is equally possible, yes. It would spare any fake names. This is straight to update, but needs more branching in the old translation function. It might be more clear however when looking at the bare data that the map is old => new. I'll change it to that, this can be part of the next review iteration.

Reply:
There are no more fake names, it's just old => new now, and new is already with namespace separators.